ompi/include/mpi.h.in

```diff
@@ -280,10 +280,35 @@
 #                    define __mpi_interface_deprecated__(msg) __attribute__((__deprecated__))
 #                endif
 #            endif
-#            if OMPI_ENABLE_MPI1_COMPAT
-#                    define __mpi_interface_removed__(msg) __mpi_interface_deprecated__(msg)
-#                    define OMPI_OMIT_MPI1_COMPAT_DECLS 0
-#            endif
+#        endif
+
+         /* For removed API, there is no portable way to cause the
+          * C compiler to error with a nice message on the usage of
+          * one of these symbols, so instead we use a C11 static_assert
+          * If the user is not using a C11 compiler, they will get an
+          * undefined reference, but no line number or nice message.
+          *
+          * Don't do MACRO magic for building Profiling library as it
+          * interferes with that system.
+          */
+#            if (OMPI_ENABLE_MPI1_COMPAT || OMPI_BUILDING)
+#                define OMPI_OMIT_MPI1_COMPAT_DECLS 0
+#                define __mpi_interface_removed__(func, newfunc) __mpi_interface_deprecated__(#func " was removed in MPI-3.0.  Use " #newfunc " instead.  continuing...")
```

Comment:
This is the only one I didn't Fix, as it was only a question.
The intent is that the deprecated warning is a non-fatal portable warning message in the build log.
If anyone adds a call to a removed function in v4.0.x ompi code, this will print a warning message during the OMPI Build, otherwise it'd just be a silent "error" (or undesirable code change).

Reply:
That's a good reason; thanks.